docs/storage-keys.rst

```diff
@@ -0,0 +1,56 @@
+####################################
+Application and request storage keys
+####################################
+
+Safir apps use the key storage on the `aiohttp.web.Application` instance to globally store data.
+Middleware also stores data in the `aiohttp.web.Request` instances.
+This page describes the conventions and keys that Safir-based apps use.
+
+Conventions
+===========
+
+Use the root app
+----------------
+
+Safir apps typically have two `~aiohttp.web.Application` instances: a root application and a sub-application.
+The sub-application serves externally-accessible routes under the application's name (``/<app-name>/``), while the root application serves internally-accessible routes relative to the root path (``/``).
+
+**Safir attaches data to the root application whenever possible.**
+
+In a request context, you can access application data, regardless of whether it is an internal or external route using the `aiohttp.web.Request.config_dict` lookup method:
+
+.. code-block:: python
+
+   @routes.get("/")
+   async def get_index(request: web.Request) -> web.Response:
+       """GET /<path>/ (the app's external root).
+       """
+       metadata = request.config_dict["safir/metadata"]
+       data = {"_metadata": metadata}
+       return web.json_response(data)
+
+safir/ key prefix
+-----------------
+
+Safir uses the ``safir/`` prefix for keys to prevent collisions with your code.
+If your application adds custom data (outside the Safir framework) to the application, consider using a custom prefix to namespace that data.
+
+Safir's application keys
+========================
+
+``safir/config``
+    The conventional key for storing your application's configuration instance.
+
+``safir/httpClient``
```

Comment:
Why camelCase instead of snake_case here? We seem to be using snake_case elsewhere.